clients/drcachesim/tools/basic_counts.cpp

```diff
@@ -257,7 +261,21 @@ basic_counts_t::print_results()
     std::cerr << "Total counts:\n";
     print_counters(total, shard_map_.size(), " total");
 
-    if (num_windows > 1) {
+    if (num_intervals > 0 && shard_map_.size() == 1) {
+        // We print per-interval counts only for serial analysis and parallel with a
+        // single shard.
+        // XXX: for parallel with multiple shards, we could merge the results based on
```

Comment:
But the user will not know what is going on: they will request an interval and just get no interval output.  I thought this parallel merging was a key aspect of this time-series feature which we needed to figure out and implement, not an afterthought.  If it's not implemented here this should be a TODO with the issue # and it should print something to the user about not being implemented and to re-run in serial mode or sthg.  But let's make sure this whole design is going to work with however the best way to merge these values is: which means we should figure that out before finalizing this interface.  Maybe these timestep absolute ordinals are what we want or maybe not and that would change the interface.

Reply:
In the previous design, the parallel merging was not a part of the upstream code. But in the current design it is (since we have the interval state snapshots now). I thought about splitting it up into a separate PR, but ended up implementing it here itself.